### ISSUE TITLE
Provide a syntax definition file with updated support for R new syntax 

### ISSUE THREAD

**Issue by cderv** [rstudio/rmarkdown#2196]
First issue has been reported in rstudio/bookdown#1157.  See discussion there for details. In a nutshell: 
* Pandoc's highlighting uses KDE XML syntax file 
* the KDE XML syntax file for R language does not correctly handles the new R syntax. 

Current fix in **rmarkdown** is for `html_document_base()` only as a post processing - see https://github.com/rstudio/rmarkdown/commit/99df4e7db5ba0108c51557639e661df05554bf1b . This fixes the issue rstudio/bookdown#1157

Proper fix should be: 

* Push upstream correct support for Pandoc for later Pandoc's version.
* Use the XML definition file directly in **rmarkdown** for previous pandoc version that won't have the updated support, using `--syntax-definition` flag.

**Comment by cderv** [comment]
From a report in https://github.com/tidyverse/reprex/issues/389, let's acknowledge that the current fix is not working for `github_document()` HTML preview

````markdown
---
title: "Minimal Example"
output: 
  github_document: default
  html_document:
    highlight: pygments
---

```{r}
library(magrittr)
1.23 %>% round()
1.23 |> round()
```
````

**Comment by cderv** [comment]
Waiting for this to be completely fixed, I published a gist with a patched xml file one could use: https://gist.github.com/cderv/079b78caefbe45da6d1c03d737970b55

It can be downloaded then used by adding `pandoc_args` in YAML header. e.g with `pdf_document()`

````
output: 
  pdf_document:
    pandoc_args: "--syntax-definition=r.xml"
````

See comment here for more detail and example on how to set a font for having the ligature in PDF

https://github.com/rstudio/bookdown/issues/1157#issuecomment-933366422

**Comment by yihui** [comment]
Could you submit a PR to KDE? I guess this is the file to edit: https://invent.kde.org/frameworks/syntax-highlighting/-/blob/master/data/syntax/r.xml The earlier we submit a PR, the better, since it will take time for the upstream change (if merged) to propagate to Pandoc.

**Comment by cderv** [comment]
I wanted to try add support for new syntax to define function too
``` r
(\(x) x + 1)(2)
#> [1] 3
```

But I still haven't done the KDE file update for this. 

You're right I should do it wit the current addition.

**Comment by cderv** [comment]
PR has been merged - they dealt with the KDE specific compilation. This means that next Pandoc version should have the new file included as they pull from the KDE repo in https://github.com/jgm/skylighting to be included in Pandoc.

**Comment by cderv** [comment]
Yes it has picked up the change and also Pandoc (https://github.com/jgm/pandoc/commit/2a70d9c3c5bd129f62fdbac0fc3c06c6aa79cc8d) but not yet in a release. Next one should have it. 

I have still a PR in skylighting for `markdown.xml` from Quarto but not sure it will be merged. 

PR for KDE with last change is ready but not yet online. I have prepare that yesterday with the changes in the PR.

**Comment by yihui** [comment]
> Yes it has picked up the change and also Pandoc ([jgm/pandoc@2a70d9c](https://github.com/jgm/pandoc/commit/2a70d9c3c5bd129f62fdbac0fc3c06c6aa79cc8d)) but not yet in a release. Next one should have it.

It seems to be shipped in Pandoc 2.18. The RStudio IDE uses Pandoc 3.x now, so perhaps we can drop `r.xml` in the #2290? (which means we just don't support syntax highlighting the pipe operators for Pandoc < 2.18 users)

**Comment by cderv** [comment]
Yes it seems ok to do that.
